internal/provider/data_source_schema_check.go

```diff
@@ -0,0 +1,137 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/labd/terraform-provider-hive/internal/sdk"
+)
+
+var _ datasource.DataSource = &HiveSchemaCheckDataSource{}
+
+func NewHiveSchemaCheckDataSource() datasource.DataSource {
+	return &HiveSchemaCheckDataSource{}
+}
+
+type HiveSchemaCheckDataSource struct {
+	client *sdk.HiveClient
+}
+
+func (r *HiveSchemaCheckDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_schema_check"
+}
+
+
+type HiveSchemaCheckDataSourceModel struct {
+	Service   types.String `tfsdk:"service"`
+	Commit    types.String `tfsdk:"commit"`
+	Author    types.String `tfsdk:"author"`
+	Schema    types.String `tfsdk:"schema"`
+	ContextId types.String `tfsdk:"context_id"`
+	Id        types.String `tfsdk:"id"`
+	Target	  types.String `tfsdk:"target"`
+	Project	  types.String `tfsdk:"target"`
```

Comment:
This conflicts with target?

Reply:
woops, thats also in the resource_schema_check. good catch.